openvpn/middlewares/client/auth/middleware.go

```diff
@@ -0,0 +1,75 @@
+package auth
+
+import (
+	log "github.com/cihub/seelog"
+	"github.com/mysterium/node/openvpn"
+	"github.com/mysterium/node/openvpn/middlewares"
+	"net"
+	"regexp"
+)
+
+type middleware struct {
+	authenticator Authenticator
+	connection    net.Conn
+	username      string
+	password      string
+	state         middlewares.State
+}
+
+// NewMiddleware creates clint user_auth challenge authentication middleware
+func NewMiddleware(authenticator Authenticator) openvpn.ManagementMiddleware {
+	return &middleware{
+		authenticator: authenticator,
+		connection:    nil,
+	}
+}
+
+func (m *middleware) Start(connection net.Conn) error {
+	m.connection = connection
```

Comment:
`connection` can be passed in in `NewMiddleware` - we'll have less mutability then.

Reply:
according to current structure, Start() is called from management.Start() method, where listener is created. ManagementMiddleware interface defines this. We cannot change without reworking all implementations.

Reply:
True, lets keep it as it is for now :)